service-statistics-unsd/src/main/java/org/oskari/statistics/plugins/unsd/UnsdDataParser.java

```diff
@@ -105,22 +117,33 @@ public StatisticalIndicatorDataDimension getTimeperiodDimensionFromIndicatorData
         }
     }
 
-    public static void parseTimePeriod(Map<Integer, Integer> countOfAreaCodesForYear, JSONObject response) {
+    public static void parseTimePeriod(Map<Integer, Set<Integer>> geoAreaCodesForYears, JSONObject response) {
         try {
             JSONArray data = response.getJSONArray(DATA_KEY);
             for (int i = 0; i < data.length(); i++) {
                 JSONObject o = (JSONObject) data.get(i);
                 Integer year = o.getInt(TIME_PERIOD_START_KEY);
-                countOfAreaCodesForYear.merge(year, 1, Integer::sum);
+                Integer geoAreaCode = o.getInt(GEO_AREA_CODE_KEY);
+
+                if(dataObjectValueTypeIsSupported(o)) {
+                    geoAreaCodesForYears.merge(year, new HashSet<Integer>(Arrays.asList(geoAreaCode)), (oldSet,newSet) -> {
```

Comment:
Here `merge` gets too complicated. More readable solution would be: `map.computeIfAbsent(year, k -> new HashSet<>()).add(geoAreaCode);`

Reply:
Updated code as suggested.